libs/build.gradle

```diff
@@ -7,7 +7,33 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-configure(childProjects.values() - project('elasticsearch-log4j')) {
+configure(childProjects.values()) {
+
+  apply plugin: 'base'
+
+  def baseProject = project
+  def baseName = "elasticsearch-${it.name}"
+  base {
+    archivesName = baseName
+  }
+  subprojects {
+    plugins.withType(JavaPlugin).whenPluginAdded {
```

Comment:
I'm not gonna lie, I struggle to follow what we're trying to do here. Can we at least add some clarifying comments here to convey not just what we're doing, but why this is necessary?

Reply:
I added comments for the entire section explaining what we are doing.